lib/webidl2.js

```diff
@@ -106,14 +147,23 @@
       if (!tokens.length) return;
       if (tokens[0].type === "whitespace") {
         const t = tokens.shift();
-        t.value.replace(/\n/g, m => {
-          line++;
-          return m;
-        });
+        let i = 0;
+        while (i !== -1) {
+          i = t.value.indexOf('\n', i);
+          if (i !== -1) {
+            ++line;
+            ++i;
+          }
+        }
         return t;
       }
     }
 
+    const all_ws_re = {
+      "ws": /([\t\n\r ]+)/y,
+      "line-comment": /\/\/(.*)\r?\n?/y,
+      "multiline-comment": /\/\*((?:[^*]|\*[^/])*)\*\//y
```

Comment:
Curious about this multiline comment regex change, is there also a performance win?

Reply:
Yes, but not huge. It's about 2% faster on V8.